Microsoft.Toolkit.Uwp.UI.Controls.WinRT/DirectWriteTextBlock/DirectWriteRenderArgBuilder.cpp

```diff
@@ -0,0 +1,298 @@
+    }
+    else
+    {
+        m_builtArgs.foregroundColor = Colors::Black;
+    }
+}
+
+void DirectWriteRenderArgBuilder::SetFontStyle(FontStyle fontStyle)
+{
+    switch (fontStyle)
+    {
+    case FontStyle::Normal: __fallthrough;
+    default:
+        m_builtArgs.fontStyle = DWRITE_FONT_STYLE::DWRITE_FONT_STYLE_NORMAL;
+        break;
+    case FontStyle::Italic:
+        m_builtArgs.fontStyle = DWRITE_FONT_STYLE::DWRITE_FONT_STYLE_ITALIC;
+        break;
+    case FontStyle::Oblique:
+        m_builtArgs.fontStyle = DWRITE_FONT_STYLE::DWRITE_FONT_STYLE_OBLIQUE;
+        break;
+    }
+}
+
+void DirectWriteRenderArgBuilder::SetFontStretch(FontStretch fontStretch)
+{
+    switch (fontStretch)
+    {
+    case FontStretch::Normal: __fallthrough;
+    default:
+        m_builtArgs.fontStretch = DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_NORMAL;
+        break;
+    case FontStretch::Condensed:
+        m_builtArgs.fontStretch = DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_CONDENSED;
+        break;
+    case FontStretch::Expanded:
+        m_builtArgs.fontStretch = DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_EXPANDED;
+        break;
+    case FontStretch::ExtraCondensed:
+        m_builtArgs.fontStretch = DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_EXTRA_CONDENSED;
+        break;
+    case FontStretch::ExtraExpanded:
+        m_builtArgs.fontStretch = DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_EXTRA_EXPANDED;
+        break;
+    case FontStretch::SemiCondensed:
+        m_builtArgs.fontStretch = DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_SEMI_CONDENSED;
+        break;
+    case FontStretch::SemiExpanded:
+        m_builtArgs.fontStretch = DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_SEMI_EXPANDED;
+        break;
+    case FontStretch::UltraCondensed:
+        m_builtArgs.fontStretch = DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_ULTRA_CONDENSED;
+        break;
+    case FontStretch::UltraExpanded:
+        m_builtArgs.fontStretch = DWRITE_FONT_STRETCH::DWRITE_FONT_STRETCH_ULTRA_EXPANDED;
+        break;
+    }
+}
+
+void DirectWriteRenderArgBuilder::SetTextOrientation(Orientation textOrientation)
+{
+    switch (textOrientation)
+    {
+    case Orientation::Vertical: __fallthrough;
+    default:
+        m_builtArgs.readingDirection = DWRITE_READING_DIRECTION::DWRITE_READING_DIRECTION_TOP_TO_BOTTOM;
+        break;
+    case Orientation::Horizontal:
+        m_builtArgs.readingDirection = DWRITE_READING_DIRECTION::DWRITE_READING_DIRECTION_LEFT_TO_RIGHT;
+        break;
+    }
+}
+
+void DirectWriteRenderArgBuilder::SetFlowDirection(FlowDirection flowDirection)
+{
+    switch (flowDirection)
+    {
+    case FlowDirection::LeftToRight: __fallthrough;
+    default:
+        m_builtArgs.flowDirection = DWRITE_FLOW_DIRECTION::DWRITE_FLOW_DIRECTION_LEFT_TO_RIGHT;
+        break;
+    case FlowDirection::RightToLeft:
+        m_builtArgs.flowDirection = DWRITE_FLOW_DIRECTION::DWRITE_FLOW_DIRECTION_RIGHT_TO_LEFT;
```

Comment:
Setting FlowDirection="RightToLeft" doesn't seem to have an effect

Reply:
I've resolved this by basically plumbing through DirectWriteReadingDirection.